### PR TITLE
Separating out the webpack public path loader and the plugin.

### PR DESCRIPTION
**Pull request by iclanton**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -162,9 +162,16 @@ engine.
 
 ## Webpack Loaders and Plugins
 
+### [@microsoft/set-webpack-public-path-plugin](./webpack/set-webpack-public-path-plugin/README.md)
+
+`set-webpack-public-path-plugin` is a plugin used to set the webpack public path variable.
```
Comment (qz2017) [inline README.md]: >plugin [](start = 38, length = 6)

webpack plugin

File: README.md [hunk 2]
```diff
@@ -162,9 +162,16 @@ engine.
...
+
+[![npm version](https://badge.fury.io/js/%40microsoft%2Fset-webpack-public-path-plugin.svg)](https://badge.fury.io/js/%40microsoft%2Fset-webpack-public-path-plugin)
+[![Dependencies](https://david-dm.org/Microsoft/set-webpack-public-path-plugin.svg)](https://david-dm.org/Microsoft/set-webpack-public-path-plugin)
+
 ### [@microsoft/loader-set-webpack-public-path](./webpack/loader-set-webpack-public-path/README.md)
 
-`loader-set-webpack-public-path` is a loader used to set the webpack public path variable.
+`loader-set-webpack-public-path` is a loader used to set the webpack public path variable. It's similar to `set-webpack-public-path-plugin`.
```
Comment (qz2017) [inline README.md]: >loader [](start = 38, length = 6)

webpack loader

File: webpack/loader-set-webpack-public-path/package.json [hunk 3]
```diff
@@ -12,13 +12,11 @@
     "test": "gulp test"
   },
   "dependencies": {
+    "@microsoft/set-webpack-public-path-plugin": "~0.0.0",
```
Comment (qz2017) [inline webpack/loader-set-webpack-public-path/package.json]: >0.0.0 [](start = 51, length = 5)

Is this a real version?

File: webpack/set-webpack-public-path-plugin/package.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "@microsoft/set-webpack-public-path-plugin",
+  "version": "0.0.0",
```
Comment (qz2017) [inline webpack/set-webpack-public-path-plugin/package.json]: >0.0.0 [](start = 14, length = 5)

why is it 0.0.0?

File: webpack/set-webpack-public-path-plugin/README.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
+# set-webpack-public-path plugin for webpack
+
+## Installation
+
+`npm @microsoft/loader-set-webpack-public-path --save-dev`
```
Comment (qz2017) [inline webpack/set-webpack-public-path-plugin/README.md]: >loader-set-webpack-public-path [](start = 16, length = 30)

wrong name

File: webpack/set-webpack-public-path-plugin/README.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+This simple plugin sets the `__webpack_public_path__` variable to
+a value specified in the arguments, optionally appended to the SystemJs baseURL
+property.
+
+# Plugin
+
+To use the plugin, add it to the `plugins` array of your Webpack config. For example:
+
+```JavaScript
+import SetPublicPathPlugin from '@microsoft/loader-set-webpack-public-path/lib/SetPublicPathPlugin';
```
Comment (qz2017) [inline webpack/set-webpack-public-path-plugin/README.md]: >loader-set-webpack-public-path [](start = 44, length = 30)

wrong name

File: webpack/set-webpack-public-path-plugin/README.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+## `getGlobalRegisterCode(bool)`
+
+This function returns a block of JavaScript that maintains a global register of script tags. If the optional boolean paramter
+is set to `true`, the code is not minified. By default, it is minified.
+
+## Usage
+
+``` javascript
+var setWebpackPublicPath = require('@microsoft/loader-set-webpack-public-path');
```
Comment (qz2017) [inline webpack/set-webpack-public-path-plugin/README.md]: >loader-set-webpack-public-path [](start = 47, length = 30)

can you update the readme? It seems a copy/paste from the other package.

**Comment by qz2017** [comment]
:clock1:

File: webpack/loader-set-webpack-public-path/package.json (hunk 3 above)
Comment (iclanton) [inline webpack/loader-set-webpack-public-path/package.json]: It should get bumped to `~1.0.0` upon publish, correct?

---
In reply to: [130456033](https://github.com/Microsoft/web-build-tools/pull/288#discussion_r130456033) [](ancestors = 130456033)

File: webpack/set-webpack-public-path-plugin/package.json (hunk 4 above)
Comment (iclanton) [inline webpack/set-webpack-public-path-plugin/package.json]: It should get bumped to `~1.0.0` upon publish, correct?

---
In reply to: [130456310](https://github.com/Microsoft/web-build-tools/pull/288#discussion_r130456310) [](ancestors = 130456310)

File: webpack/set-webpack-public-path-plugin/README.md (hunk 7 above)
Comment (iclanton) [inline webpack/set-webpack-public-path-plugin/README.md]: It's not, I just missed a few things in the updates.

---
In reply to: [130456600](https://github.com/Microsoft/web-build-tools/pull/288#discussion_r130456600) [](ancestors = 130456600)

**Pull request merged**